torch/distributed/device_mesh.py

```diff
@@ -385,10 +386,14 @@ class DeviceMesh:
         represented as a n-d dimension array, and each value of the n-d dimensional
         array is the global id of the default process group ranks.
 
-        DeviceMesh could be used to describe the layout of devices across the cluster,
-        and serves as a proxy for communication among the device lists within the cluster.
+        DeviceMesh could be used to setup the N dimensional device connections across the cluster,
+        and manage the ProcessGroups for N dimensional parallelisms. Communications could happen on
+        each dimension of the DeviceMesh separately. DeviceMesh respect the device that user select
+        already (i.e. if user call `torch.cuda.set_device` before the DeviceMesh initialization),
+        and will select/set the device for the current process if user does not set the device
+        beforehands. Note that manual device selection should happen BEFORE the DeviceMesh initialization.
 
-        DeviceMesh can be used as a context manager.
+        DeviceMesh can also be used as a context manager when using together with most DTensor APIs.
```

Comment:
Curious, what are the APIs that this doesn't work?

Reply:
iirc every public DTensor API should work, I just want to be conservative here. All non-DTensor APIs would probably not work because the context manager is just stashing the device mesh created to the mesh env, nothing magic